docs/src/doc/docs/developer_guide/architecture/data_model/extra.md

```diff
@@ -0,0 +1,97 @@
+# Extra
+
+## Introduction
+
+`ExtraProperty` classes are used both by [Documentable](documentables.md) and [Content](page_content.md#content-model)
+models.
+
+Source code for `ExtraProperty`:
+
+```kotlin
+interface ExtraProperty<in C : Any> {
+    interface Key<in C : Any, T : Any> {
+        fun mergeStrategyFor(left: T, right: T): MergeStrategy<C> = MergeStrategy.Fail {
+            throw NotImplementedError("Property merging for $this is not implemented")
+        }
+    }
+
+    val key: Key<C, *>
+}
+```
+
+To declare a new extra, you need to implement `ExtraProperty` interface. It is advised to use following pattern
+when declaring new extras:
+
+```kotlin
+data class CustomExtra(
+    [any data relevant to your extra], 
+    [any data relevant to your extra] 
+): ExtraProperty<Documentable> {
+    override val key: CustomExtra.Key<Documentable, *> = CustomExtra
+    companion object : CustomExtra.Key<Documentable, CustomExtra>
+}
+```
+
+Merge strategy for extras is invoked only if merged objects have different values for same Extra.
```

Comment:
`different` here is used as `each has their own`, not as `not equals`, but I agree it can be confusing. That actually migrated from [old documentation](https://kotlin.github.io/dokka/1.7.0-SNAPSHOT/developer_guide/data_model/#extraproperty-class-usage)

Rephrased it a little, hopefully it's better now